### PR TITLE
Fix carrierwave 'exists?'

### PR DESCRIPTION
**Pull request by samphilipd**
CarrierWave `exists?` method is broken. It should use the public_id, not the identifier.

P.S. My editor auto-strips whitespace, hence the large diff.

### CODE REVIEW

File: lib/cloudinary/carrier_wave.rb [hunk 1]
```diff
@@ -157,35 +157,35 @@ def initialize(identifier, uploader)
...
     def delete
       public_id = @resource_type == "raw" ? self.filename : self.public_id
-      Cloudinary::Uploader.destroy(public_id, :type=>self.storage_type, :resource_type=>self.resource_type) if @uploader.delete_remote?        
+      Cloudinary::Uploader.destroy(public_id, :type=>self.storage_type, :resource_type=>self.resource_type) if @uploader.delete_remote?
     end
-    
+
     def exists?
-      Cloudinary::Uploader.exists?(self.identifier, :type=>self.storage_type, :resource_type=>self.resource_type)
+      Cloudinary::Uploader.exists?(self.public_id, :type=>self.storage_type, :resource_type=>self.resource_type)
```
Comment (samphilipd) [inline lib/cloudinary/carrier_wave.rb]: Only change of note is here.

**Comment by morganick** [comment]
Any news on this? This is still broken.

**Comment by samphilipd** [comment]
@morganick I gave up waiting for this PR, instead we simply stopped using Cloudinary at work and switched to S3, mostly because this gem is so bad.

**Comment by morganick** [comment]
@samphilipd no worries, my comment was to the guys at cloudinary. I was in the process of upgrading some gems until I hit my head on this. It does not look good when this fix has been sitting here for over a year. It's, evidently, not the only thing broken when you upgrade carrierwave either. 😞

**Comment by taragano** [comment]
Sorry for the delay everyone,
This issue should be fixed now. See:
https://github.com/cloudinary/cloudinary_gem/commit/dd8fd69d9406545defda46598ef511e516766157
Please update to the latest version and try again?

**Comment by tocker** [comment]
@samphilipd thanks for the pull request! 
I'm closing it as we applied a different solution. Please retest and let us know if the issue persists.

**Pull request closed without merging**